### PR TITLE
Refactoring of GLPK-based approximating simplex solver #6968

### PR DESCRIPTION
**Pull request by aytey**
This PR implements the changes as discussed in #6968 to clean-up the interface around the `ApproxGLPK` class.

Namely:

- Minimised the interface around `ApproximateSimplex` to expose only what's used by the rest of `cvc5`
- Change most methods in `ApproximateSimplex` to be pure virtual (some methods were `static`, but we actually had a class instance, so there are now instance methods, to allow `ApproximateSimplex` to be an interface class); `mkApproximateSimplexSolver` and `enabled` are left as `static`
- All method bodies and members that were in `ApproximateSimplex` are now in `ApproxGLPK`
- The class `ApproxNoOp` has been removed
- Slight reordering of the code to provide a more logical flow, now that most of the code is under `#ifdef CVC5_USE_GLPK`, and to allow for `ApproxGLPK` to be declared at the top of `approx_simplex.cpp`
- Anything that remains "ordered weirdly" is probably an artefact from renaming `ApproximateSimplex` -> `ApproxGLPK`

Minor bits:

- Ran `git-clang-format`
- Removed `using namespace std` (and added `std::` where required) to ensure consistency in `approx_simplex.cpp`

Signed-off-by: Andrew V. Teylu <andrew@tey.lu>

### CODE REVIEW

**Comment by andrewvaughanj** [comment]
Given GLPK isn't tested via CI, I am not sure what is the best way to ensure that these changes are "okay"; maybe we should wait until #6958 is merged and we have GLPK in the nightlies before merging this?

File: src/theory/arith/approx_simplex.cpp
Comment (andrewvaughanj) [reply src/theory/arith/approx_simplex.cpp]: Done!

File: src/theory/arith/approx_simplex.cpp
Comment (andrewvaughanj) [reply src/theory/arith/approx_simplex.cpp]: Done!

File: src/theory/arith/approx_simplex.cpp
Comment (andrewvaughanj) [reply src/theory/arith/approx_simplex.cpp]: Done!

File: src/theory/arith/approx_simplex.h
Comment (andrewvaughanj) [reply src/theory/arith/approx_simplex.h]: Done!

File: src/theory/arith/approx_simplex.cpp
Comment (andrewvaughanj) [reply src/theory/arith/approx_simplex.cpp:984]: I'm wondering if I should uncomment these and change them to be `Debug("approx")` prints -- what do you think?

Basically, I think some of these lines could be useful (even though they're currently commented-out).

Comment (andrewvaughanj) [reply src/theory/arith/approx_simplex.cpp]: Okay, I did this. We can revert the commit if you prefer to remove this code instead.

**Review by 4tXJ7f — APPROVED** [review]
LGTM, thanks for the updates and sorry about the late review.

**Comment by andrewvaughanj** [comment]
@nafur, @ajreynol what should we do about the debugging information that was removed in https://github.com/cvc5/cvc5/pull/7576? Are you happy for me to reintroduce the bits that I feel shouldn't have been removed (and help with debugging/working with the GLPK integration)?

**Comment by nafur** [comment]
> @nafur, @ajreynol what should we do about the debugging information that was removed in #7576? Are you happy for me to reintroduce the bits that I feel shouldn't have been removed (and help with debugging/working with the GLPK integration)?

Yes, please reintroduce them if you feel they help debugging. My main motivation for removing them was that I wanted to get rid of `Notice()` macro. Just use `Trace` instead.

**Comment by andrewvaughanj** [comment]
> > @nafur, @ajreynol what should we do about the debugging information that was removed in #7576? Are you happy for me to reintroduce the bits that I feel shouldn't have been removed (and help with debugging/working with the GLPK integration)?
> 
> Yes, please reintroduce them if you feel they help debugging. My main motivation for removing them was that I wanted to get rid of `Notice()` macro. Just use `Trace` instead.

Okay, I can (and will) do this 🎉

**Comment by aytey** [comment]
@4tXJ7f: okay, this is now rebased with the debugging information re-introduced (as discussed).

**Comment by aytey** [comment]
🥳

**Pull request merged**
